trillian/integration/ct_hammer/main.go

```diff
@@ -93,43 +110,111 @@ func main() {
 	if err != nil {
 		glog.Exitf("Failed to read log config: %v", err)
 	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
-	var caChain, leafChain []ct.ASN1Cert
-	var signer crypto.Signer
-	var leafCert, caCert *x509.Certificate
-	if *testDir != "" {
-		// Retrieve the test data.
-		caChain, err = integration.GetChain(*testDir, "int-ca.cert")
+	var generatorFactory func(c *configpb.LogConfig) (integration.ChainGenerator, error) // Almost Java-esque...
+	if len(*srcLogURI) > 0 || len(*srcLogName) > 0 {
```

Comment:
This code is getting quite unwieldy. Maybe split some of it out and have two blocks of code that return a suitable generator?

Reply:
So you want a `generatorFactoryGenerator`?  Done.

Reply:
More like a Provider<CertGenerator> in Guice so it could be injected. Have to make do in Go though :-)